roles/kubernetes/master/templates/kubeadm-config.v1alpha2.yaml.j2

```diff
@@ -2,7 +2,8 @@ apiVersion: kubeadm.k8s.io/v1alpha2
 kind: MasterConfiguration
 api:
 {% if groups['kube-master'] | length > 1 and kubeadm_config_api_fqdn is defined %}
-  controlPlaneEndpoint: {{ kubeadm_config_api_fqdn }}:{{ loadbalancer_apiserver.port | default(kube_apiserver_port) }}
+  controlPlaneEndpoint: {{ kubeadm_config_api_fqdn }}
+  bindPort: {{ loadbalancer_apiserver.port | default(kube_apiserver_port) }}
```

Comment:
https://github.com/kubernetes/kubernetes/blob/release-1.11/cmd/kubeadm/app/apis/kubeadm/v1alpha2/types.go#L108
According to the description here, I feel that both of them are ok.
what do you think?
Are you sure you have an error when deploy?

Reply:
cc @neolit123

Reply:
> According to the description here, I feel that both of them are ok.

assuming that `kubeadm_config_api_fqdn` does not include a port, i think with or without this PR it should be the same, unless there is a bug somewhere.

here is how `controlPlaneEndpoint` is parsed:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/endpoint.go#L83-L111

@whereismyjetpack 
could you please give your input values and show the errors that kubeadm prints.